lib/node_modules/@stdlib/array/bool/test/test.filter.js

```diff
@@ -0,0 +1,188 @@
+var Uint8Array = require( '@stdlib/array/uint8' );
+var BooleanArray = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof BooleanArray, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'attached to the prototype of the main export is a `filter` method', function test( t ) {
+	t.strictEqual( hasOwnProp( BooleanArray.prototype, 'filter' ), true, 'has property' );
+	t.strictEqual( isFunction( BooleanArray.prototype.filter ), true, 'has method' );
+	t.end();
+});
+
+tape( 'the method throws an error if invoked with a `this` context which is not a boolean array instance', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new BooleanArray( 5 );
+
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[],
+		function noop() {}
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.filter.call( value, predicate );
+		};
+	}
+
+	function predicate( v ) {
+		return ( v === true );
+	}
+});
+
+tape( 'the method throws an error if provided a first argument which is not a function', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new BooleanArray( 10 );
+
+	values = [
+		'5',
+		3.14,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[]
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.filter( value );
+		};
+	}
+});
+
+tape( 'the method returns an empty array if operating on an empty boolean array', function test( t ) {
+	var arr;
+	var out;
+
+	arr = new BooleanArray();
+	out = arr.filter( predicate );
+
+	t.strictEqual( out.length, 0, 'returns expected value' );
+	t.end();
+
+	function predicate( v ) {
+		return ( v === true );
+	}
+});
+
+tape( 'the method returns a new boolean array containing only those elements which satisfy a test condition', function test( t ) {
+	var expected;
+	var actual;
+	var arr;
+
+	arr = new BooleanArray( [ true, false, true ] );
+	expected = new Uint8Array( [ 1, 1 ] );
+	actual = arr.filter( predicate );
+
+	t.strictEqual( instanceOf( actual, BooleanArray ), true, 'returns expected value' );
+	t.strictEqual( actual.length, expected.length, 'returns expected value' );
+	t.deepEqual( reinterpretBoolean( actual, 0 ), expected, 'returns expected value' );
+	t.end();
+
+	function predicate( v ) {
+		return ( v === true );
+	}
+});
+
+tape( 'the method copies all elements to a new array if all elements satisfy a test condition', function test( t ) {
+	var expected;
+	var actual;
+	var arr;
+
+	arr = new BooleanArray( [ true, false, true, false ] );
+	expected = new Uint8Array( [ 1, 1 ] );
```

Comment:
```suggestion
	arr = new BooleanArray( [ true, true, true, true ] );
	expected = new Uint8Array( [ 1, 1, 1, 1 ] );
```

The test case should match the test description at L145.